### PR TITLE
ccmlib/node.py: refactor stress() and stress_object()

### PR DESCRIPTION
**Pull request by fruch**
since `stress()` was returning only stderr and stdout
`stress_object()` treated anything printed in the stderr as errors
but cassandra stress also print warning into the stderr
and `stress_object()` started accumlating all kind of strings
it should ignore.

this change align those two function to be working similar
to the code in the ccm upstream, and now `stress()` returns
a nametuple that has rc,stderr,stdout. so the user can read
the outout and parse it as it wants, if the `rc!==0` an exception
is being raised (caller should catch it, if it's expected to fail
non of the dtest use it like that)

Test: added an integration test to cover those two function

Close: #402
Ref: scylladb/scylladb#11445

### CODE REVIEW

**Review by fee-mendes — APPROVED** [review]
LGTM, I've ran it locally and the "failling"  test now got through. :-)

File: tests/conftest.py [hunk 1]
```diff
@@ -76,7 +76,9 @@ def relocatable_cluster(test_dir, test_id):
         'range_request_timeout_in_ms': timeout,
         'write_request_timeout_in_ms': timeout,
         'truncate_request_timeout_in_ms': timeout,
-        'request_timeout_in_ms': timeout
+        'request_timeout_in_ms': timeout,
+        'skip_wait_for_gossip_to_settle': 0,
+        'ring_delay_ms': 0
```
Comment (bhalevy) [inline tests/conftest.py:81]: This seems to do more than refactoring.
Better extract it to a separate patch and document the motivation

File: ccmlib/node.py [hunk 2]
```diff
@@ -2022,3 +2027,17 @@ def make_pat_for_log_level(level):
...
+        out = out.decode()
+    if (err is not None) and isinstance(err, bytes):
+        err = err.decode()
+    rc = process.returncode
+
+    if rc != 0:
+        raise ToolError(cmd_args, rc, out, err)
+
+    ret = namedtuple('Subprocess_Return', 'stdout stderr rc')
+    return ret(stdout=out, stderr=err, rc=rc)
```
Comment (bhalevy) [inline ccmlib/node.py:2043]: Won't the break dtests that use stress and expect a `(stdout, stderr)` tuple to be returned?

If ccm was a submodule in scylla-dtest we could adjust the callers when updating the submodule,
but today the integration is at runtime.

I think this api change deserves a new name for the function, while keeping the old one backward compatible.
Maybe after converting all users we can rename the new function back to `stress` and get rid of the old interface.

**Review by bhalevy — CHANGES_REQUESTED** [review]

**Review by bhalevy — CHANGES_REQUESTED** [review]
sorry, I meant clicking `Request changes`

File: ccmlib/node.py (hunk 2 above)
Comment (fruch) [reply ccmlib/node.py:2043]: I've checked the user code in dtests, and we don't need, no test is using the return values from those.

File: tests/conftest.py (hunk 1 above)
Comment (fruch) [reply tests/conftest.py:81]: I've added this so I can run the unittest quicker, (we can't enable it by default in dtest, but those integration test, this probably good enough)

File: ccmlib/node.py (hunk 2 above)
Comment (fruch) [reply ccmlib/node.py:2043]:
```bash
scylla-dtest on  fix_wrong_loggers:master [🏎💨] 🐍 took 2s
🟢 ❯ grep -inr '\.stress(' ./
./incremental_repair_test.py:48:        node1.stress(['write', 'n=10000', '-schema', 'replication(factor=3)'])
./incremental_repair_test.py:129:        node1.stress(['write', 'n=10000', '-schema', 'replication(factor=2)'])
./incremental_repair_test.py:144:        node2.stress(['write', 'n=15000', '-schema', 'replication(factor=2)'])
./incremental_repair_test.py:228:        node1.stress(['write', 'n=5M', '-rate', 'threads=10', '-schema', 'replication(factor=3)'])
./incremental_repair_test.py:283:        node1.stress(['write', 'n=3', '-rate', 'threads=1', '-schema', 'replication(factor=3)'])
./secondary_indexes_test.py:797:        node1.stress(['write', 'n=50K', 'no-warmup'])
./secondary_indexes_test.py:916:        node.stress(['write', 'n=50K', 'no-warmup', '-schema', 'replication(factor=3)',
./secondary_indexes_test.py:2198:        node.stress(['write', 'n=50K', 'no-warmup', '-schema', 'replication(factor=3)',
./sstablesplit_test.py:39:        node.stress(['write', 'n=10000', '-rate', 'threads=10'])
./sstablesplit_test.py:47:        node.stress(['read', 'n=10000', '-rate', 'threads=25'])
./sstablesplit_test.py:110:        node.stress(['write', 'n=20000', '-rate', 'threads=10',
./repair_test.py:65:        node1_1.stress(stress_options=['write', 'n=50K', 'cl=ONE', '-schema', 'replication(factor=2)'])
./repair_test.py:85:        node1_1.stress(stress_options=['write', 'n=50K', 'cl=ONE', '-schema', 'replication(factor=2)'])
./repair_test.py:101:        node1.stress(stress_options=['write', 'n=50K', 'cl=ONE', '-schema', 'replication(factor=3)'])
./repair_test.py:117:        node1_1.stress(stress_options=['write', 'n=50K', 'cl=ONE', '-schema', 'replication(factor=2)'])
./repair_test.py:492:        self.node1.stress(stress_options=['write', 'n=5000', 'cl=ONE', '-schema', 'replication(factor=3)'])
./concurrent_schema_changes_test.py:557:            node1.stress(args)
./encryption_at_rest_test.py:65:        node1.stress(['write', 'n=100000', 'cl=QUORUM', '-rate', 'threads=8'])
./encryption_at_rest_test.py:68:        node1.stress(['read', 'n=100000', 'cl=QUORUM', '-rate', 'threads=8'])
./snapshot_test.py:362:            results, errors = node.stress(['write', 'duration=5m', '-mode', 'cql3', 'native', '-rate', 'threads=100', '-pop', 'seq=1..100000000', '-log', 'interval=5'],
./snapshot_test.py:397:        results, errors = node1.stress(['write', 'n=10000', '-rate', 'threads=10'],
./snapshot_test.py:429:        results, errors = node1.stress(['write', 'n=1000000', '-rate', 'threads=10'],
./update_cluster_layout_tests.py:303:        node1.stress(['write', 'n=20000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks1'])
./update_cluster_layout_tests.py:304:        node2.stress(['write', 'n=20000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks2'])
./update_cluster_layout_tests.py:305:        node3.stress(['write', 'n=20000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks3'])
./update_cluster_layout_tests.py:372:        node1.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks1'])
./update_cluster_layout_tests.py:373:        node2.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks2'])
./update_cluster_layout_tests.py:374:        node3.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks3'])
./update_cluster_layout_tests.py:444:        node1.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks1'])
./update_cluster_layout_tests.py:445:        node2.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks2'])
./update_cluster_layout_tests.py:446:        node3.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks3'])
./update_cluster_layout_tests.py:534:        node1.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks1'])
./update_cluster_layout_tests.py:535:        node2.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks2'])
./update_cluster_layout_tests.py:1494:        node1.stress(
./update_cluster_layout_tests.py:1671:        node1.stress(['user', 'n=%s' % nr_partitions, 'cl=ONE', 'profile=%s' % c_s_profile, 'ops(insert=1)',
./update_cluster_layout_tests.py:2361:            node2.stress(['write', 'cl=QUORUM',  'n=%s' % n, 'no-warmup',
./update_cluster_layout_tests.py:2370:        node2.stress(['read', 'cl=ONE', 'n=%s' % n, 'no-warmup', '-pop seq=1..%s' % n, '-rate threads=20'])
./topology_test.py:50:        node1.stress(['write', 'n=10K', '-rate', 'threads=8'])
./persistence_test.py:38:        node.stress(['write', 'n=5', "no-warmup", "cl=ALL", "-pop",
./persistence_test.py:43:        node.stress(['read', 'n=5', "no-warmup", "cl=ALL", "-pop", "seq=1...5",
./batch_test.py:556:        node.stress(["user", "no-warmup", "profile=%s" % os.path.realpath('test_data/batch-test/complex_schema.yaml'),
./batch_test.py:558:                     # node.stress(["user", "no-warmup", "profile=/tmp/complex_schema.yaml", "ops(insert=1)", "cl=ALL",
./sstable_generation_loading_test.py:105:        node1.stress(['write', 'n=10000', '-rate', 'threads=8'])
./cql_additional_tests.py:6621:            results, errors = node.stress(['write', 'duration=15s', '-mode', 'cql3', 'native', '-rate', 'threads=50', '-pop', 'seq=1..100000000', '-log', 'interval=5'],
./data_distribution_balance_test.py:41:        self.cluster.stress(stress_cmd.split(" "))
./jmxmetrics_test.py:154:            node.stress(['write', 'n=100K'])
./commitlog_test.py:144:        self.node1.stress(['write', 'n=150000', '-rate', 'threads=25'])
./commitlog_test.py:193:            self.node1.stress(
./commitlog_test.py:888:        cs_task = executor.submit(lambda: node1.stress(cs_cmd, capture_output=True))
./nodetool_additional_test.py:2122:        return self.stress(node, 'write', times=times, duration=duration, col=col, pop=pop, opt=opt, cl=cl, expected_errors=expected_errors)
./nodetool_additional_test.py:2127:        return self.stress(node, 'mixed', times=times, duration=duration, col=col, pop=pop, opt=opt, expected_errors=expected_errors)
./nodetool_additional_test.py:2443:            node2.stress(
./stress_tool_test.py:45:        node.stress(['write', 'n=1000', '-rate', 'threads=50', '-col', 'n=FIXED(50)',
./materialized_views_test.py:240:        stdout, stderr = node1.stress(stress_options=['write', 'cl=QUORUM', 'n={}'.format(n),
./tools/stress.py:51:        node.stress(cs_cmdline)
./scyllatop_test.py:121:        node.stress(['write', 'duration=10s', "no-warmup", '-rate', 'threads=2'])
./scyllatop_test.py:142:        node.stress(['write', 'duration=10s', "no-warmup", '-rate', 'threads=2'])
./nodetool_test.py:71:        cs_result, cs_err = node.stress(
./manager_repair_tests.py:556:        nodes[1].stress(stress_command)
./replace_address_test.py:103:        node1.stress(['write', 'n=10000', '-schema', 'replication(factor=3)'])
./replace_address_test.py:296:        node1.stress(['write', 'n=10000', '-schema', 'replication(factor=3)'])
./replace_address_test.py:362:        node1.stress(['write', f'n={keys}', '-schema', 'replication(factor=3)'])
./replace_address_test.py:410:        node1.stress(['write', f'n={keys}', '-schema', 'replication(factor=3)'])
./replace_address_test.py:475:        node1.stress(["write", "n=10000", "-schema", "replication(factor=2)"])
./replace_address_test.py:548:            node1.stress(['write', 'duration=320s', 'no-warmup', 'cl=QUORUM', '-rate',
./replace_address_test.py:610:        node1.stress(['write', 'n=10000', '-schema', 'replication(factor=1)'])
./jmx_test.py:44:        node1.stress(["write", "n=15M", "-schema", "replication(factor=3)", "-rate", "threads=50", ])
./jmx_test.py:82:        node1.stress(["write", "n=500K", "-schema", "replication(factor=3)", ])
./jmx_test.py:119:        node1.stress(["write", "n=10K", "-schema", "replication(factor=3)", ])
./jmx_test.py:173:        node.stress(["write", "n=1", ])
./jmx_test.py:177:        node.stress(["write", "n=750K", ])
./migration_test.py:462:        node1.stress(['user', 'profile={}'.format(profile_path), 'ops(insert=1)', 'n={}'.format(stress_count), '-rate',
./largecolumn_test.py:18:        node.stress(['write', 'n=5', "no-warmup", "cl=ALL", "-pop", "seq=1...5", "-schema",
./largecolumn_test.py:20:        node.stress(['read', 'n=5', "no-warmup", "cl=ALL", "-pop", "seq=1...5", "-schema",
./largecolumn_test.py:78:        node1.stress(['write', 'n=10000', "no-warmup", "-col", "n=fixed(1)", "size=fixed(40000)", "-rate", "threads=2"])
./largecolumn_test.py:79:        node1.stress(['mixed', "no-warmup", 'duration=15s', '-pop', 'seq=1..10000',
./simple_no_driver_test.py:32:        node.stress(['write', 'n=100000', '-mode', 'cql3', 'simplenative',
./simple_no_driver_test.py:43:            node.stress(['read', 'n=100000', '-mode', 'cql3', 'simplenative', '-rate', 'threads=1', '-pop',
./cql_tracing_test.py:299:        stdout, stderr = node1.stress(stress_options=['write', 'cl=ONE', 'n=10000',
./compaction_additional_test.py:636:        node1.stress(['write', 'n=0', 'no-warmup', '-schema', 'replication(factor=1)', '-rate', 'threads=1'])
./compaction_additional_test.py:705:            node1.stress(['read', 'n=100', 'no-warmup', '-rate', 'threads=10', '-col', 'size=FIXED(1024)'])
./rolling_upgrade_test.py:116:        return executor.submit(lambda: node.stress(stress_options=stress_command, capture_output=True))
./in_memory_test.py:391:        node1.stress(stress_options=['write', 'n=%s' % num_keys, 'cl=ALL', '-schema', 'replication(factor=3)'])
./in_memory_test.py:398:            node1.stress(stress_options=['read', 'n=%s' % num_keys, 'cl=ONE', '-schema', 'replication(factor=3)'])
./sstableutil_test.py:104:        node.stress(['write', 'n=1', '-rate', 'threads=1'])
./sstableutil_test.py:107:        node.stress(['write', f'n={numrecords}', '-rate', 'threads=50'])
./sstableutil_test.py:112:        node.stress(['read', f'n={numrecords}', '-rate', 'threads=25'])
./backup_restore_tests.py:722:        node1.stress(['read', 'n=1000', "no-warmup", '-rate', 'threads=2', "-pop", "seq=1...1000"])
./backup_restore_tests.py:742:        node.stress(['write', f'n={n}', "no-warmup", '-rate', 'threads=2', "-pop",
./compaction_test.py:787:        node.stress(['write', 'n=0', 'no-warmup', '-schema', 'replication(factor=1)', '-rate', 'threads=1'])
./compaction_test.py:924:    node.stress(['write', f'n={keycount}'])
./hintedhandoff_additional_test.py:506:        node2.stress(['write', f'n={rows}', '-schema', 'replication(factor=3)'])
./cqlsh_tests/cqlsh_copy_tests.py:1255:                self.node1.stress(['write', f'n={num_operations} cl=ALL', 'no-warmup', '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1258:                self.node1.stress(['user', f'profile={profile}', 'ops(insert=1)',
./cqlsh_tests/cqlsh_copy_tests.py:1446:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1483:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1519:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1557:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1594:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_copy_tests.py:1630:        self.node1.stress(['write', 'n={}'.format(num_records), '-rate', 'threads=50'])
./cqlsh_tests/cqlsh_tests.py:1846:        self.cluster.nodelist()[0].stress(['write', 'n=1K', '-rate', 'threads=8'])
./updateable_config_test.py:64:        node1.stress(['write', 'n=10000', '-rate', 'threads=8'])
./updateable_config_test.py:72:        node1.stress(['mixed', 'n=10000', '-rate', 'threads=8'])
./updateable_config_test.py:75:        node1.stress(['mixed', 'n=10000', '-rate', 'threads=8'])
./updateable_config_test.py:151:        node1.stress(['write', 'n=10000', '-rate', 'threads=8'])
./ics_compaction_test.py:227:            results, errors = node1.stress(stress_params, capture_output=True)
./metadata_tests.py:32:        node1.stress(['read', 'no-warmup', 'n=30000', '-schema', 'replication(factor=2)', 'compression=LZ4Compressor',
./metadata_tests.py:53:            node1.stress(['write', 'no-warmup', 'n=30000', '-schema', 'replication(factor=2)',
./manager_backup_tests.py:186:        healthy_node.stress(['read', f'n={number_of_rows}', '-rate', f'threads={threads}'])
./manager_backup_tests.py:375:        self.cluster.stress(
./manager_backup_tests.py:484:        self.cluster.stress(['write', 'cl=ALL', 'n=5000K', '-rate', 'threads=50', '-schema', 'replication(factor=4)'])
./manager_backup_tests.py:552:        self.cluster.stress(['write', 'n=10000K', '-rate', 'threads=50',
./manager_backup_tests.py:673:        self.cluster.stress(['write', 'n=5000K', '-rate', 'threads=50',
./manager_backup_tests.py:770:        self.cluster.stress(['write', 'n=50K', '-rate', 'threads=50', '-pop', 'seq=1..100000'])
./manager_backup_tests.py:777:            self.cluster.stress(['write', 'n=50K', '-rate', 'threads=50', '-pop',
./bootstrap_test.py:265:        node1.stress(['write', 'n=10000', '-rate', 'threads=8'])
./bootstrap_test.py:303:        node1.stress(['write', 'n=100000', '-schema', 'replication(factor=2)'])
./bootstrap_test.py:346:        node1.stress(['write', 'n=100000', '-schema', 'replication(factor=2)'])
./bootstrap_test.py:386:        node1.stress(['write', 'n=1000', '-schema', 'replication(factor=2)',
./bootstrap_test.py:437:        node1.stress(['user', 'profile=' + stress_config.name, 'n=2000000',
./bootstrap_test.py:445:            node1.stress(['user', 'profile=' + stress_config.name, 'ops(insert=1)',
./bootstrap_test.py:481:        node1.stress(['write', 'n=10000', '-rate', 'threads=8'])
./bootstrap_test.py:542:        node1.stress(['write', 'n=10K', '-rate', 'threads=8'])
./bootstrap_test.py:593:        node1.stress(['write', 'n=100000', '-rate', 'threads=8'])
./bootstrap_test.py:653:        node1.stress(['write', 'n=500K', '-schema', 'replication(factor=1)',
./bootstrap_test.py:765:        stress_thread = executor.submit(lambda: node1.stress(stress_options=write_stress_cmd, capture_output=True))
./rest_api_tests.py:98:        self.cluster.stress(['write'] + stress_cmd)
./rest_api_tests.py:99:        self.cluster.stress(['read'] + stress_cmd)
./offline_tools_test.py:74:        node1.stress(['write', 'n=0', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:91:        node1.stress(['write', 'n=1K', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:104:        node1.stress(['write', 'n=50K', '-rate', 'threads=20', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:170:        node1.stress(['write', 'n=1', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:184:        node1.stress(['write', 'n=1K', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:195:        node1.stress(['write', 'n=100K', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:242:        node1.stress(['write', 'n=100', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:248:        node1.stress(['write', 'n=10K', '-schema', 'replication(factor=1)'])
./offline_tools_test.py:250:        node1.stress(['write', 'n=10K', '-schema', 'replication(factor=1)'])
```

Comment (bhalevy) [reply ccmlib/node.py:2043]: let's do that in the same next promotion window then

**Review by bhalevy — APPROVED** [review]

File: ccmlib/node.py (hunk 2 above)
Comment (fruch) [reply ccmlib/node.py:2043]: addressing it in:
https://github.com/scylladb/scylla-dtest/pull/2853
